src/lightning_app/core/api.py

```diff
@@ -317,14 +316,18 @@ async def api_catch_all(request: Request, full_path: str):
 fastapi_service.mount("/static", StaticFiles(directory=frontend_static_dir, check_dir=False), name="static")
 
 
-# Catch-all for frontend routes, must be defined after all other routes
-@fastapi_service.get("/{full_path:path}", response_class=HTMLResponse)
 async def frontend_route(request: Request, full_path: str):
     if "pytest" in sys.modules:
         return ""
     return templates.TemplateResponse("index.html", {"request": request})
 
 
+def register_global_routes():
+    # Catch-all for nonexistent API routes (since we define a catch-all for client-side routing)
+    fastapi_service.get("/api{full_path:path}", response_class=JSONResponse)(api_catch_all)
```

Comment:
Is this idempotent?
Running `register_global_routes` multiple times is allowed?